.github/workflows/tests.yml

```diff
@@ -0,0 +1,45 @@
+name: brew test-bot
```

Comment:
Both `tests.yml` and `publish.yml` were initially provided by brew — I kept `test.yml` mostly the same, whereas I refactored `publish.yml` so we preserve historical versions and also changed how it's triggered. But the origin of these two workflows were scaffoled by brew when run running the `brew tap-new knocklabs/homebrew-tap` command (which I followed from this [blog post](https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/)).

To see the initial scaffolded workflows by brew, check it out here:
https://github.com/knocklabs/homebrew-tap/commit/d91d75802a714f1f37d302477ccfbb96c860312f